### ISSUE TITLE
Proc/Lambda case operator

### ISSUE THREAD

**Issue by wied03** [opal/opal#1278]
Thought I had filed this already but can't find it.

code:

```ruby
def foo 
  !(lambda { 3 } === lambda { 4 })
  rescue
    true
end

puts foo
```

MRI:
```
true
```

Opal master, `bundle exec opal -A test.rb`:
```
false
```

**Comment by iliabylich** [comment]
Looks like there's an issue with arity checking for `lambda`, it's not related to the case comparison:
``` ruby
l = lambda { 1 }
puts l.call(25)

# MRI:
# wrong number of arguments (1 for 0) (ArgumentError)

# Opal master with -A:
# 1
```

But that's [expected](https://github.com/opal/opal/blob/master/opal/corelib/proc.rb#L40) - in JS `Function.prototype.apply` and `Function.prototype.call` don't check arity. For Opal methods it's made on the compilation phase, but lambda is just a JS function that got `call`-ed. Here's what you get by compiling `lambda { 1 }`:
``` ruby
($a = ($b = self).$lambda, $a.$$p = (TMP_1 = function(){var self = TMP_1.$$s || this;

  return 1}, TMP_1.$$s = self, TMP_1), $a).call($b)
```

Not sure that this issue can be fixed

**Comment by iliabylich** [comment]
@elia Is it possible to append arity checking to `lambda` on the compilation stage? I know, it's just a `Kernel` method, but I guess there's no cross-browser way to detect arity of a JS function (not even talking about splats/kwargs)

**Comment by elia** [comment]
I think we could store arity on the function itself (`$$arity`?) and then have it checked or not depending on the configuration…

**Comment by wied03** [comment]
@iliabylich - I didn't do a thorough root cause on this one, was just making sure that any last issues discovered with opal-rspec stuff were accounted for.

**Comment by wied03** [comment]
This is fixed as of 0.10.0.rc.1
